src/lib.rs

```diff
@@ -10,6 +10,24 @@
 
 //! A typesafe bitmask flag generator.
 
+// Compile the crate with no_std if possible. The "no_std" feature can be
+// removed once no_std becomes available in 1.6.0 stable. In the meantime no_std
+// must be disabled by default to allow the crate to work on older rust versions.
+#![cfg_attr(feature = "no_std", no_std)]
+
+#[cfg(all(feature = "no_std", test))]
+#[macro_use]
+extern crate std;
```

Comment:
Something like this?
```rust
// Compile the crate with no_std if possible. The "no_std" feature can be
// removed once no_std becomes available in 1.6.0 stable. In the meantime no_std
// must be disabled by default to allow the crate to work on older rust versions.
#![cfg_attr(feature = "no_std", no_std)]

#[cfg(all(feature = "no_std", test))]
#[macro_use]
extern crate std;

#[cfg(not(feature = "no_std"))]
extern crate std as core;

// Re-export libcore using an alias so that the macros can work in no_std crates
// while remaining compatible with normal crates.
#[doc(hidden)]
pub use core as __core;
```